src/main/python/snakepit/main.py

```diff
@@ -56,6 +57,13 @@ class TemplateNoteFoundException(Exception):
     pass
 
 
+class OutputDirectoryNotWritable(Exception):
```

Comment:
Rather than checking first, you should probably `try` `except` when attempting to write.